README.md

```diff
@@ -1,3 +1,262 @@
+
+### Row key filter push down
+
+This connector supports pushing down some of the filters on the row key column
+in the DataFrame to Bigtable
+and performing them on the server-side. The list of supported or non-supported
+filters is as follows:
+
+| Filter               | Push down filter supported |
+|----------------------|----------------------------|
+| `EqualTo`            | Yes                        |
+| `LessThan`           | Yes                        |
+| `GreaterThan`        | Yes                        |
+| `LessThanOrEqual`    | Yes                        |
+| `GreaterThanOrEqual` | Yes                        |
+| `StringStartsWith`   | Yes                        |
+| `Or`                 | Yes                        |
+| `And`                | Yes                        |
+| `Not`                | No                         |
+| Compound Row Key     | No                         |
+
+Note that when using compound row keys, filter on those columns are
+**not** pushed to Bigtable and are performed on the client-side (resulting in a
+full-table scan).
+
+### Client-side metrics
+
+Since the Bigtable Spark connector is based on the
+[Bigtable Client for Java](https://github.com/googleapis/java-bigtable),
+client-side metrics are enabled
+inside the connector by default. You can refer to the
+[client-side metrics](https://cloud.google.com/bigtable/docs/client-side-metrics)
+documentation to find more details on accessing and interpreting these metrics.
+
+### Use low-level RDD functions with Bigtable
+
+Since the Bigtable Spark connector is based on the
+[Bigtable client for Java](https://github.com/googleapis/java-bigtable),
+you can directly use the client in your Spark applications and perform
+distributed read or write requests
+within the low-level RDD functions such as `mapPartitions`
+and `foreachPartition`.
+
+To use the Bigtable client for Java classes, append the
+`com.google.cloud.spark.bigtable.repackaged` prefix to the package names. For
+example, instead of using the class name
+as `com.google.cloud.bigtable.data.v2.BigtableDataClient`, use
+`com.google.cloud.spark.bigtable.repackaged.com.google.cloud.bigtable.data.v2.BigtableDataClient`.
+
+## Examples
+
+You can access examples for Java, Scala, and Python inside the `examples`
+directory. Each directory contains a `README.md` file with instruction on
+running the example inside.
+
+## Limitations
+
+Currently, only some types are supported in the connector catalog, as follows:
+
+| Type                  | Catalog support |
+|-----------------------|-----------------|
+| `long`                | Yes             |
+| `string`              | Yes             |
+| `binary` (byte array) | Yes             |
+
+Support for other types, e.g., `int`, `float`, etc., will be added in future
+versions of the
+connector. The `examples` folder contains workaround for converting these types
+to `BinaryType`
+inside your application, in different languages. For complex types,
```

Comment:
Additionally, the columns may be converted to a supported type, e.g. string(<column>), in SQL when writing and converted back to the intended type on read.